plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -267,6 +278,73 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, constraint.getSummary()));
     }
 
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupingSets)
+    {
+        if (!isAggregationPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        PinotTableHandle tableHandle = (PinotTableHandle) handle;
+
+        if (tableHandle.getQuery().isPresent()) {
+            return Optional.empty();
+        }
+        if (tableHandle.getLimit().isPresent()) {
+            // handle's limit is applied after aggregations, so we cannot apply aggregations if limit is already set
+            return Optional.empty();
```

Comment:
Can this be modelled with `io.trino.plugin.pinot.query.DynamicTable#query` as a nested query?

See https://github.com/trinodb/trino/blob/6e3bc7857690ea14bcd122881a703cc7440793f7/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java#L248

and the test
https://github.com/trinodb/trino/blob/f2582d1537a21e4422d4b477a198ef472838231a/plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java#L520

Reply:
Updated - did not completely remove the dynamic table class, but now filters and limits can be applied multiple times and there are tests to verify pushdown.

Reply:
Will update the interface in a subsequent pr